web/crux-ui/src/server/crux/image-service.ts

```diff
@@ -126,33 +137,126 @@ class DyoImageService {
-  return !config
-    ? null
-    : {
-        ...config,
-        networkMode: networkModeToDto(config.networkMode),
-      }
+  if (!config) return undefined
+
+  const explicit: ExplicitContainerConfig = {
+    user: config.user ?? null,
+    tty: config.TTY ?? false,
+    ports: config.ports ?? [],
+    portRanges: config.portRanges ?? [],
+    volumes: config.volumes ?? [],
+    commands: config.command ?? [],
+    args: config.args ?? [],
+    expose: config.expose ?? null,
+    ingress: config.ingress ?? null,
+    configContainer: config.configContainer ?? null,
+    importContainer: config.importContainer ?? null,
+    logConfig: logConfigToDto(config.dagent?.logConfig),
+    restartPolicy: restartPolicyTypeToDto(config.dagent?.restartPolicy),
+    networkMode: networkModeToDto(config.dagent?.networkMode),
+    deploymentStrategy: deploymentStrategyToDto(config.crane?.deploymentStatregy),
+    customHeaders: config.crane?.customHeaders ?? [],
+    proxyHeaders: config.crane?.proxyHeaders ?? false,
+    useLoadBalancer: config.crane?.useLoadBalancer ?? false,
+    extraLBAnnotations: config.crane?.extraLBAnnotations ?? null,
+    healthCheckConfig: config.crane?.healthCheckConfig ?? null,
+    resourceConfig: config.crane?.resourceConfig ?? null,
+  }
+
+  return explicit
 }
 
 export const explicitContainerConfigToProto = (config?: ExplicitContainerConfig): ProtoExplicitContainerConfig => {
-  return !config
-    ? undefined
-    : {
-        user: config?.user ?? undefined,
-        expose: config?.expose ?? undefined,
-        mounts: config?.mounts ?? [],
-        ports: config?.ports ?? [],
-        networkMode: networkModeToProto(config.networkMode),
-      }
+  if (!config) return null
+
+  const protoConfig: ProtoExplicitContainerConfig = {
+    user: config.user,
+    TTY: config.tty,
+    ports: config.ports ?? [],
+    portRanges: config.portRanges ?? [],
+    volumes: (config.volumes ?? []) as Volume[],
+    command: config.commands ?? [],
+    args: config.args ?? [],
+    importContainer: config.importContainer,
+    configContainer: config.configContainer,
+    dagent: undefined,
+    crane: undefined,
+  }
+
+  if (config.expose) {
+    protoConfig.expose = { public: config.expose.public, tls: config.expose.tls }
+  }
+
+  if (config.ingress) {
+    protoConfig.ingress = {
+      name: config.ingress.name,
+      host: config.ingress.host,
+      uploadLimit: config.ingress.uploadLimitInBytes,
+    }
+  }
+
+  if (config.logConfig || config.restartPolicy || config.networkMode) {
+    protoConfig.dagent = {
+      logConfig: logConfigToProto(config.logConfig),
+      restartPolicy: restartPolicyTypeToProto(config.restartPolicy),
+      networkMode: networkModeToProto(config.networkMode),
+    }
+  }
+
+  if (
+    config.deploymentStrategy ||
+    config.customHeaders ||
+    config.proxyHeaders ||
+    config.useLoadBalancer ||
+    config.healthCheckConfig ||
+    config.resourceConfig ||
```

Comment:
If for example `config.healthCheckConfig` is true than 
```
resourceConfig: config.resourceConfig
```
will fail, not? Becuase `config.resourceConfig` is `undefined`.

Reply:
All of these properties are optional. This statement's purpose is that i want to hold the crane undefined unless one of its property is defined. 
Why do you think it will fail?

Reply:
Okey, nvm. I didn't see the optional.